src/main/java/com/ace/alfox/controllers/ActionController.java

```diff
@@ -16,19 +18,28 @@
 @RestController
 class ActionController {
 
+    @Autowired
+    private AutowireCapableBeanFactory beanFactory;
+
+    @Autowired
+    private PlayerFactory playerFactory;
+
     @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
     public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
         HttpSession session = request.getSession(true);
         String playerId = (String) session.getAttribute("cid");
 
-        Player player = PlayerFactory.fetch(playerId);
+        Player player = playerFactory.fetch(playerId);
         if(player == null) {
             return ResponseEntity.notFound().build();
         }
 
         ActionFactory doSomething = ActionFactory.named(command).withParameters(params);
+        // Alright, so we've got the IAction in ActionFactory but, because we instantiated it Spring hasn't done any
+        // dependency injection, so we're going to hint that it should do it now.
+        beanFactory.autowireBean(doSomething.action);
```

Comment:
The "wart" in my opinion.

Reply:
Yeah we definitely shouldn't have to do this, especially here in the controller. We can likely find a better way :)

Reply:
I think once I remove all static functions in `ActionFactory` I'll be able to inject the `beanFactory` right into `ActionFactory` and perform the autowire right [after initialization](https://github.com/project-alfox/backend/pull/3/files/48f077e50210c16f8a80bcfde27eb5bec1134587#diff-0bb83ee27a66a987201591144090b4ecR44)